pkg/account/component.go

```diff
@@ -132,17 +133,47 @@ func (c *component) UpdatePassword(ctx context.Context, currentPassword, newPass
 	var userID = ctx.Value(cs.CtContextUserID).(string)
 	var username = ctx.Value(cs.CtContextUsername).(string)
 
-	if currentPassword == newPassword || newPassword != confirmPassword {
-		return errorhandler.Error{
-			Status:  http.StatusBadRequest,
-			Message: keycloakb.ComponentName + "." + "invalidValues",
+	if currentPassword == "" {
+		// Current password is empty, meaning the user wants to set their password. Then, we need to check they do not have a password credential
+		credentialsKc, err := c.keycloakAccountClient.GetCredentials(accessToken, realm)
+		if err != nil {
+			c.logger.Warn(ctx, "msg", "Can't get credentials", "err", err.Error())
+			return err
 		}
-	}
 
-	_, err := c.keycloakAccountClient.UpdatePassword(accessToken, realm, currentPassword, newPassword, confirmPassword)
-	if err != nil {
-		c.logger.Warn(ctx, "msg", "Can't update password", "err", err.Error())
-		return err
+		for _, cred := range credentialsKc {
+			if *cred.Type == "password" {
+				c.logger.Warn(ctx, "msg", "Can't set password", "err", "user has a password credential")
```

Comment:
I think "Can't set initial password as user already owns one" would be clearer ("err" is not a mandatory tag)